canvasapi/current_user.py

```diff
@@ -150,3 +153,168 @@ def get_bookmark(self, bookmark):
             'users/self/bookmarks/{}'.format(bookmark_id)
         )
         return Bookmark(self._requester, response.json())
+
+    def get_favorite_courses(self, **kwargs):
+        """
+        Retrieve the paginated list of favorite courses for the current user.
+        If the user has not chosen any favorites,
+        then a selection of currently enrolled courses will be returned.
+
+        :calls: 'GET /api/v1/users/self/favorites/courses \
+        <https://canvas.instructure.com/doc/api/favorites.html#method.favorites.list_favorite_courses>'_
+
+        :rtype: :class:`canvasapi.paginated_list.PaginatedList`
+        of :class:`canvasapi.course.Course`
+        """
+
+        return PaginatedList(
+            Course,
+            self._requester,
+            'GET',
+            'users/self/favorites/courses',
+            _kwargs=combine_kwargs(**kwargs)
+        )
+
+    def get_favorite_groups(self, **kwargs):
+        """
+        Retrieve the paginated list of favorite groups for the current user.
+        If the user has not chosen any favorites, then a selection of groups
+        that the user is a member of will be returned.
+
+        :calls: 'GET /api/v1/users/self/favorites/courses \
+        <https://canvas.instructure.com/doc/api/favorites.html#method.favorites.list_favorite_groups>'_
+
+        :rtype: :class:`canvasapi.paginated_list.PaginatedList`
+        of :class:`canvasapi.group.Group`
+        """
+
+        return PaginatedList(
+            Group,
+            self._requester,
+            'GET',
+            'users/self/favorites/groups',
+            _kwargs=combine_kwargs(**kwargs)
+        )
+
+    def add_favorite_course(self, course, **kwargs):
+        """
+        Add a course to the current user's favorites. If the course is already
+        in the user's favorites, nothing happens.
+
+        :calls: 'POST /api/v1/users/self/favorites/courses/:id \
+        <https://canvas.instructure.com/doc/api/favorites.html#method.favorites.add_favorite_course>'_
+
+        :param course: The course or ID/SIS ID of the course.
+        :type course: :class:`canvasapi.course.Course` or int
+
+        :rtype: :class:`canvasapi.favorite.Favorite`
+        """
+
+        course_id = obj_or_id(course, "course", (Course,))
```

Comment:
Because the documentation explicitly states that this value can be an SIS ID, we should consider allowing string types in certain cases. Perhaps a similar pattern to `Canvas.get_user` or `Canvas.get_course`?